lib/service-data/service-data.js

```diff
@@ -75,6 +75,15 @@ serviceData.getInstancesByType = (query, {nested = false} = {}) => {
   ).filter((instanceType) => instanceType !== 'page.admin')
 }
 
+/*
+ *  We delete a property by passing `undefined` as the third argument to `setInstanceProperty`
+ *
+ *  Here, there is no third argument -- it is implicit that the third argument to `setInstanceProperty`
+ *  is undefined -- ensuring that the property is deleted, rather than calling `setInstanceProperty`
+ *  and (accidently) supplying a valid value
+ */
+serviceData.deleteInstanceProperty = (_id, propertyKey) => serviceData.setInstanceProperty(_id, propertyKey)
```

Comment:
I decided not to put this into Runner, but keep it in Editor:
- Runner doesn't use it (or, Runner _may_ delete properties but there isn't an existing method, and it would be madness to go looking for named-but-undefined variables)
- Editor already creates other methods which are missing from the collection it imports from Runner (see `countInstancesByType`, below)
- `setInstanceProperty` in Runner is 120 lines of untested whoah which I don't want to duplicate/refactor right now
- The context where `deleteInstanceProperty` is used makes it clear that this is a delete

Reply:
(It's also now clearer where this function is used whether a delete or a change is occurring)

Reply:
i think this is good. i don't see why runner needs to modify the schema it should treat the schema as a readonly object. editor should be responsible for modifying it.

Reply:
Yeah ... you'd think that, but _who knows_